roomserver/internal/input_latest_events.go

```diff
@@ -292,7 +345,12 @@ func (u *latestEventsUpdater) makeOutputNewRoomEvent() (*api.OutputEvent, error)
 	for _, entry := range u.stateBeforeEventAdds {
 		ore.StateBeforeAddsEventIDs = append(ore.StateBeforeAddsEventIDs, eventIDMap[entry.EventNID])
 	}
-	ore.SendAsServer = u.sendAsServer
+	// If we are overwriting the latest events and state then we don't
+	// want to send out any changes that happened as a result to servers
+	// over federation.
+	if !u.stateAtEvent.Overwrite {
+		ore.SendAsServer = u.sendAsServer
```

Comment:
Can you put this with `:237-:239` for clarity please.

Reply:
Do you mean just to mention the line numbers in the comment here?

Reply:
Actually, these lines may not be necessary anyway because of `237-239`.